jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/test/java/org/jbpm/workbench/cm/client/list/AbstractCaseInstanceListTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.workbench.cm.client.list;
+
+import java.util.List;
+
+import com.google.common.collect.Lists;
+import org.jboss.errai.common.client.api.Caller;
+import org.jbpm.workbench.cm.model.CaseInstanceSummary;
+import org.jbpm.workbench.cm.service.CaseManagementService;
+import org.jbpm.workbench.cm.util.CaseInstanceSearchRequest;
+import org.jbpm.workbench.cm.util.CaseStatus;
+import org.junit.Before;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.uberfire.client.mvp.PlaceManager;
+import org.uberfire.mocks.CallerMock;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+public class AbstractCaseInstanceListTest {
```

Comment:
Introducing this type of class seems redundant when already having this class available ```org.jbpm.workbench.cm.client.util.AbstractCaseInstancePresenterTest```. Most of what you need is already there, you can just add the mocks and parts in init() method which you need.
Re-using this class instead will also make you rethink the approach about injecting mocks in a superclass. I strongly recommend moving it to the designated test class, even if it means duplicating injection across test classes. The price is not too big and in return we will have easier to understand test classes.

Reply:
Another point is that it is mixing view and presenter test in here, I would also recommend leaving the presenter test as is and just add the necessary code for the view test class.

Reply:
Ok, I do not know the abstract test class. AbstractCaseInstancePresenterTest have many sub class(IMO, it is base class of all test of caseinstancePresenter), I think we should not add more CaseInstancePresenterList's logic in here. I would recommend let AbstractCaseInstanceListTest
extends it, move common logic(like: CaseManagementService caseManagementService;) into it.

Reply:
@cristianonicolai I find we can not extends AbstractCaseInstancePresenterTest, it's presenter is AbstractCaseInstancePresenter total different to AbstractPresenter.

Reply:
@bxf12315 maybe try to keep it simple first, just add a new test class...